lightning/src/blinded_path/utils.rs

```diff
@@ -135,17 +135,49 @@ fn encrypt_payload<P: Writeable>(payload: P, encrypted_tlvs_rho: [u8; 32]) -> Ve
 	write_adapter.encode()
 }
 
-/// Blinded path encrypted payloads may be padded to ensure they are equal length.
-///
-/// Reads padding to the end, ignoring what's read.
-pub(crate) struct Padding {}
+/// Represents optional padding for encrypted payloads.
+/// Padding is used to ensure payloads have a consistent length.
+pub(crate) struct Padding {
+	padding: Option<Vec<u8>>
```

Comment:
It wold be preferable if we didn't need to allocate a `Vec`. Could we simply store the size and have the `Writeable` implementation write the appropriate number of bytes?

Reply:
Makes sense!
Update the `Padding` to simply contain the `usize` instead of `Vec`. [pr3177.02](https://github.com/shaavan/rust-lightning/commits/pr3177.02)

I was not able to find a direct implementation of `Writeable for Bytes`, so I took inspiration from `Readable for Padding` to update the `Writable`.

Let me know if the implementation looks good! Thanks a lot!